codec-http/src/main/java/io/netty/handler/codec/http/HttpHeadersBuilder.java

```diff
@@ -0,0 +1,308 @@
+     * </a>.
+     * When disabling this validation, it is the responsibility of the caller to ensure that the values supplied
+     * do not contain a non-url-escaped carriage return (CR) and/or line feed (LF) characters.
+     *
+     * @param validation If validation should be enabled or disabled.
+     * @return The new builder.
+     */
+    public HttpHeadersBuilder withValueValidation(boolean validation) {
+        return withValueValidator(validation ? DEFAULT_VALUE_VALIDATOR : NO_VALUE_VALIDATOR);
+    }
+
+    /**
+     * Create a new builder that with the given {@link ValueValidator}.
+     * <p>
+     * <b>Warning!</b> If the given validator does not check that the header values are standards compliant, Netty won't
+     * validate & protect against user-supplied headers that are malicious.
+     * This can leave your server implementation vulnerable to
+     * <a href="https://cwe.mitre.org/data/definitions/113.html">
+     *     CWE-113: Improper Neutralization of CRLF Sequences in HTTP Headers ('HTTP Response Splitting')
+     * </a>.
+     * When disabling this validation, it is the responsibility of the caller to ensure that the values supplied
+     * do not contain a non-url-escaped carriage return (CR) and/or line feed (LF) characters.
+     *
+     * @param validator The HTTP header name validator to use.
+     * @return The new builder.
+     */
+    public HttpHeadersBuilder withValueValidator(ValueValidator<CharSequence> validator) {
+        if (valueValidator == checkNotNull(validator, "validator")) {
+            return this;
+        }
+        if (notSubclassed() && nameValidator == DEFAULT_NAME_VALIDATOR && validator == DEFAULT_VALUE_VALIDATOR) {
+            return combiningHeaders ? DEFAULT_COMBINING : DEFAULT;
+        }
+        return with(nameValidator, validator, combiningHeaders);
+    }
+
+    /**
+     * Create a new builder that has HTTP header validation enabled or disabled.
+     * <p>
+     * <b>Warning!</b> Setting {@code validation} to {@code false} will mean that Netty won't
+     * validate & protect against user-supplied headers that are malicious.
+     * This can leave your server implementation vulnerable to
+     * <a href="https://cwe.mitre.org/data/definitions/113.html">
+     *     CWE-113: Improper Neutralization of CRLF Sequences in HTTP Headers ('HTTP Response Splitting')
+     * </a>.
+     * When disabling this validation, it is the responsibility of the caller to ensure that the values supplied
+     * do not contain a non-url-escaped carriage return (CR) and/or line feed (LF) characters.
+     *
+     * @param validation If validation should be enabled or disabled.
+     * @return The new builder.
+     */
+    public HttpHeadersBuilder withValidation(boolean validation) {
+        if (notSubclassed()) {
+            return validation ? DEFAULT : DEFAULT_NO_VALIDATION;
+        }
+        return withNameValidation(false).withValueValidation(false);
```

Comment:
Should it be `validation` instead of `false`?

Reply:
Yes, well spotted.